DOC-12785 Add a note for the vector index restoration process (#3935)
couchbase/docs-server: modules/manage/pages/manage-backup-and-restore/manage-backup-and-restore.adoc

You can restore a backup of data from a bucket using the Couchstore storage engine to one using Magma.
You can also restore a Magma-backed bucket backup to a Couchstore bucket.

[NOTE]
====

> When restoring Vector Indexes, you must run the BUILD command only after the Key-Value Data restoration is complete.

Reviewer: I think this should be BUILD INDEX 

```suggestion
When restoring Vector Indexes, you must run the BUILD INDEX command only after the Key-Value Data restoration is complete.
```

> The sequence of stages in the restoration process is same as that of the xref:backup-restore:cbbackupmgr-backup.adoc#DISCUSSION[backup] process.

Reviewer: Wording tweaks

```suggestion
The sequence of stages in the restoration process is the same as that of the xref:backup-restore:cbbackupmgr-backup.adoc#DISCUSSION[backup] process.
```

> Starting the build before all the data is restored can result in incomplete codebook training or an inaccurate codebook.

Reviewer 2: "can result in incomplete codebook training or an inaccurate codebook. This can cause training failures or unreliable query results."

my suggestion should be to write, "can result in training failure or inaccurate codebook generation. This can cause build to be errored out or unreliable query results."

Author: For clarity, changed it to:

"can result in the training failure or inaccurate codebook.
This can cause the build to either error-out or produce unreliable query results."

This can cause training failures or unreliable query results.

> Also, the re-training of codebook after the build has started is not yet supported.

Reviewer 3: suggest replacing with:
The re-training of codebook after the build has started is not supported.

Author: "Also" connects the current and the previous sentences. So, retaining it.

> Also, the re-training of codebook after the build has started is not supported.

Reviewer: Wording tweaks

```suggestion
Starting the build before all the data is restored can result in training failure or an inaccurate codebook.
This can cause the build to either error-out or produce unreliable query results.
Also, re-training the codebook after the build has started is not supported.
```

Couchbase recommends waiting for the full dataset restoration.

> Only after the restoration is complete, run a build to make sure that the codebook generated out of build accurately represents the completely restored dataset.

Reviewer 3: Run a build once the restoration is complete to make sure that the codebook....the completely restored dataset.

Author: The sequence in which I have written is correct. We don't want to confuse the reader.
